test/cleanup.test.js

```diff
@@ -1,187 +1,284 @@
-        return { stdout: args[3], stderr: null };
-      } else if(args[3] === 'dddddddddddddddddddddddddddddddddddddddd') {
-        return { stdout: args[3], stderr: null };
-      }else if (args[1] === 'tag') {
-        return { stdout: null, stderr: null };
-      }
-      break;
-    case 'grep':
-      if (args[2] === gitMergeCommitStdout) {
-        return { stdout: '2', stderr: null };
-      } else if (args[1] === 'tag' && args[0] === 'dddddddddddddddddddddddddddddddddddddddd') {
-        return { stdout: 'tag', stderr: null };
-      } else if ((args[0] === 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' || args[0] === 'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb') && args[1]) {
-        t.equals(args[1], args[0]);
-        return { stdout: args[0], stderr: null };
-      } else {
-        return { stdout: null, stderr: null };
+
+
+const exec = {
+  restore: () => child_process.exec.restore(),
+
+  mock: () => {
+    return sinon.stub(child_process, 'exec', (cmd, callback) => {
+      const commandParts = cmd.split(' ');
+      const command = commandParts[2];
+
+      const sha = {
+        'cat-file': commandParts[4],
+        tag: commandParts[5],
+        'rev-parse': commandParts[4]
+      }[command];
+
+      let stdout = '';
+
+      switch (command) {
+      case 'cat-file':
+        if (sha === 'cccccccccccccccccccccccccccccccccccccccc' ||
+            sha === 'eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee') {
+          stdout = '2'; // these are merge commits
+        } else {
+          stdout = '0';
+        }
+
+        break;
+
+      case 'tag':
+        if (sha === 'dddddddddddddddddddddddddddddddddddddddd')
+          stdout = sha; // this one is a tag
+
+        break;
+
+      case 'rev-parse':
+        if (sha !== 'this-is-an-image-i-named-myself')
+          stdout = sha; // everything is a commit except this one image tag
+
+        break;
       }
 
-    }
-  };
-  require('child_process').spawnSync = spawns;
-  t.end();
-});
+      callback(null, stdout);
+    });
+  }
+};
 
-test('handleCb, error', (t) => {
-  let stub = sinon.stub(process, 'exit');
-  let log = sinon.stub(console, 'log');
-  file.handleCb(error);
-  console.log.restore();
-  t.equal(stub.getCall(0).args[0], 1, 'error should exit 1');
-  t.equal(log.getCall(0).args[0], error, 'should log error');
-  process.exit.restore();
-  t.end();
-});
+test('getImages, error', (assert) => {
+  const stub = AWS.stub('ECR', 'describeImages').yields(new Error('foo'));
 
-test('handleCb, error', (t) => {
-  let stub = sinon.stub(process, 'exit');
-  let log = sinon.stub(console, 'log');
-  file.handleCb(null, success);
-  console.log.restore();
-  t.equal(stub.getCall(0).args[0], 0, 'success should exit 0');
-  t.equal(log.getCall(0).args[0], success, 'should log success');
-  process.exit.restore();
-  t.end();
-});
+  cleanup.getImages(region, repo, (err) => {
+    assert.ok(
+      stub.calledWith({ repositoryName: repo }),
```

Comment:
`.calledWith()` is a great part of sinon's API -- far easier to read than `stub.getCall(0).args[0]`.